### PR TITLE
Detect sample files that are missing when delivering to customer inbox on caesar

### PR DESCRIPTION
**Pull request by henningonsbring**
## Description
This PR adds a function which will report the samples that won't be sent to caesar during delivery This will increase the transparency for production when something goes wrong and will make it easier to communicate with customers.

This information regarding missing samples is needed for later PRs when adding support for sending an e-mail to production for completed rsyncs via cron tab

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh missing-files`

### How to test
- [x] do `cg deliver ticket -d fastq -t 821682`
- [x] do `housekeeper delete file -y <ID>` for ID = [IDs for one sample from ticket 821682]
- [x] do `cg deliver ticket -d fastq -t 821682` (i.e. `cg deliver ticket` when data for one sample is missing)

### Expected test outcome
- [x] check that "Data has been delivered for all samples" is in the output when no data is missing
- [x] check that sample 21ET100319 is missing after running the `housekeeper delete file -y <ID>` command
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [x] tests executed by @henningonsbring 
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by henningonsbring** [comment]
"Data has been delivered for all samples" in output when data for all samples included

```
cg deliver ticket -d fastq -t 821682

[part of output removed to reduce length of this message]

Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L002_R2_001.fastq.gz
Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L003_R2_001.fastq.gz
Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L004_R2_001.fastq.gz
Data has been delivered for all samples
Creating folder: /home/proj/stage/rsync/821682_210616_13_37_15_670509
Submit sbatch /home/proj/stage/rsync/821682_210616_13_37_15_670509/821682_rsync.sh
Running command sbatch /home/proj/stage/rsync/821682_210616_13_37_15_670509/821682_rsync.sh
Submitted batch job 1312375
Rsync to caesar running as job 1312375
```

**Comment by henningonsbring** [comment]
21ET100319 reported as missing when sample removed from housekeeper

```
cg deliver ticket -d fastq -t 821682

[part of output removed to reduce length of this message]

Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L002_R2_001.fastq.gz
Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L003_R2_001.fastq.gz
Removing file: /home/proj/stage/customers/cust056/inbox/821682/21ET100326/HVH7TDSXY_21ET100326_S289_L004_R2_001.fastq.gz
No data delivered for sample(s):
21ET100319
Creating folder: /home/proj/stage/rsync/821682_210616_13_39_58_811311
Submit sbatch /home/proj/stage/rsync/821682_210616_13_39_58_811311/821682_rsync.sh
Running command sbatch /home/proj/stage/rsync/821682_210616_13_39_58_811311/821682_rsync.sh
Submitted batch job 1312376
Rsync to caesar running as job 1312376
```

File: cg/meta/deliver_ticket.py [hunk 1]
```diff
@@ -91,6 +91,36 @@ def remove_files(self, reads: List[Path]) -> None:
             LOG.info("Removing file: %s", file)
             file.unlink()
 
+    def get_all_samples_from_ticket(self, ticket_id: int) -> list:
+        all_samples = []
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        for case in cases:
+            for link_obj in case.links:
+                all_samples.append(link_obj.sample.name)
+        return all_samples
```
Comment (Mropat) [inline cg/meta/deliver_ticket.py:100]: similar functionality exists:
https://github.com/Clinical-Genomics/cg/blob/7b5366430ec63860dcbb1ed2df236af9707710f5/cg/store/api/findbusinessdata.py#L223

if its not good enough, you could create one in the store api

**Review by Mropat — APPROVED** [review]
Left minor comment, otherwise approved!

**Comment by henningonsbring** [comment]
@Mropat addressed your comment and tested again on stage (it worked), lets merge and deploy

**Comment by Mropat** [comment]
```
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 22.8.6
    Uninstalling cg-22.8.6:
      Successfully uninstalled cg-22.8.6
Successfully installed cg-22.9.0
```

**Pull request merged**